### ISSUE TITLE
Epoch number is lost after resume

### ISSUE THREAD

**Issue by emjotde** [rsennrich/nematus#2]
When training has been interrupted and is later resumed the number of iterations is correctly displayed, however, the epoch number is lost and does not seem to increase later on during training.

**Comment by bastings** [comment]
This is a bit tricky. Even if you include the epoch number into the saved model, the code would start running from a newly shuffled data set, which means that its epoch idx would not truly reflect the number of epochs. 

I think the best way to solve this is to just compute the number of epochs based the number of data batches and the number of updates.

**Comment by bastings** [comment]
By the way, the first number requires going through the (training) data once.

Data iterator can be modified like this:

```python
self.num_batches = sum(1 for _ in self) 
```

**Comment by rsennrich** [comment]
closing this.

since it's quite normal to resume training with a different training set, I think the current behavior (counting the number of epochs since restarting) isn't too bad, even though no behavior will be optimal for all use cases.
